docs/guide/docs/advanced/create-channel.md

```diff
@@ -0,0 +1,110 @@
+Channels are the gateway between Botpress and an external communication service.
```

Comment:
This PR includes other changes we had to do while working on the feature. Since we had to refactor channels to support the new format of contents for prompts, I started the doc here.